hopsworks-common/src/main/java/io/hops/hopsworks/common/featurestore/trainingdatasets/external/ExternalTrainingDatasetController.java

```diff
@@ -16,45 +16,47 @@
 
 package io.hops.hopsworks.common.featurestore.trainingdatasets.external;
 
+import com.google.common.base.Strings;
 import io.hops.hopsworks.common.dao.featurestore.storageconnector.s3.FeaturestoreS3Connector;
 import io.hops.hopsworks.common.dao.featurestore.trainingdataset.TrainingDataset;
 import io.hops.hopsworks.common.dao.featurestore.trainingdataset.external.ExternalTrainingDataset;
 import io.hops.hopsworks.common.featurestore.FeaturestoreConstants;
+import io.hops.hopsworks.common.featurestore.storageconnectors.FeaturestoreStorageConnectorType;
 import io.hops.hopsworks.common.featurestore.storageconnectors.s3.FeaturestoreS3ConnectorFacade;
+import io.hops.hopsworks.common.featurestore.trainingdatasets.TrainingDatasetDTO;
 import io.hops.hopsworks.exceptions.FeaturestoreException;
 import io.hops.hopsworks.restutils.RESTCodes;
 
 import javax.ejb.EJB;
 import javax.ejb.Stateless;
 import javax.ejb.TransactionAttribute;
 import javax.ejb.TransactionAttributeType;
+import java.nio.file.Paths;
 import java.util.logging.Level;
 
 /**
  * Class controlling the interaction with the external_training_dataset table and required business logic
  */
 @Stateless
+@TransactionAttribute(TransactionAttributeType.NEVER)
 public class ExternalTrainingDatasetController {
   @EJB
   private ExternalTrainingDatasetFacade externalTrainingDatasetFacade;
   @EJB
   private FeaturestoreS3ConnectorFacade featurestoreS3ConnectorFacade;
-  
+
+
   /**
-   * Persists an external training dataset
-   *
-   * @param externalTrainingDatasetDTO the user input data to use when creating the external training dataset
-   * @return the created training dataset
+   * Create and persist an external training dataset
+   * @param connector
+   * @param path
+   * @return
    */
-  @TransactionAttribute(TransactionAttributeType.NEVER)
-  public ExternalTrainingDataset createExternalTrainingDataset(ExternalTrainingDatasetDTO externalTrainingDatasetDTO)
-      throws FeaturestoreException {
-    // Verify external training datasset specifc input
-    FeaturestoreS3Connector featurestoreS3Connector =
-      verifyExternalTrainingDatasetS3ConnectorId(externalTrainingDatasetDTO.getS3ConnectorId());
-    
+  @TransactionAttribute(TransactionAttributeType.REQUIRED)
+  public ExternalTrainingDataset createExternalTrainingDataset(FeaturestoreS3Connector connector, String path) {
     ExternalTrainingDataset externalTrainingDataset = new ExternalTrainingDataset();
-    externalTrainingDataset.setFeaturestoreS3Connector(featurestoreS3Connector);
+    externalTrainingDataset.setFeaturestoreS3Connector(connector);
```

Comment:
Yep. They were there because they were using some controllers. I made them slimmer and now moved them in the facades. I'm not sure we even need these tbh. I think the EntityManager should be able to write them together with the `TrainingDataset` object.